libp2p/Session.cpp

```diff
@@ -229,14 +231,19 @@ void Session::send(bytes&& _msg)
 void Session::write()
 {
     bytes const* out = nullptr;
+
+    bool disconnectImminent = false;
```

Comment:
This can be simplified to:

bool disconnectImminent = (writeQueue[0][0] == DisconnectPacket);

Also, since the type of the disconnectImminent value is obvious, we can make disconnectImmiment auto (and const since we're not going to modify its value after initially setting it). 

However, you can remove the disconnectImmiment check completely since @gumb0 brought up a good way to simplify the changes - start the timer in Session::disconnect after the sealAndSend call. The resulting wait won't be exactly 2 seconds after the disconnect packet is actually sent out but it should be close enough.